src/test/java/io/vertx/core/net/impl/pool/ConnectionPoolTest.java

```diff
@@ -335,12 +365,21 @@ public void testRemoveSingleConnectionWithWaiter() throws Exception {
 
   @Test
   public void testConnectFailureWithPendingWaiter() throws Exception {
+    // This is necessary because:
```

Comment:
@vietj is it supposed to happen?

Reply:
yes it can happen why ?

Reply:
In this case the test isn't written assuming that post actions can happen in different moments, modifying the state not atomically (look at the comments I've added past this `//`)